src/App.tsx

```diff
@@ -1,56 +1,62 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'debounce';
+import classNames from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './PeopleList';
+import { getCheckQuery } from './helper/getCheckQuery';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [people] = useState<Person[]>(peopleFromServer);
+  const [hasOpen, setHasOpen] = useState(false);
```

Comment:
The hasOpen state variable name is not descriptive. Consider renaming to isDropdownOpen for better readability.